### DISCUSSION TITLE
CMAKE does not use Visual Studio 2019 Professional

### DISCUSSION THREAD

**Discussion by RazaelDAbyss** [o3de/o3de#7892] [Help Wanted] [answered]
When using the following command line instruction from the Build the Engine portion of the "Setting up O3DE from GitHub" tutorial
cmake -B build/windows_vs2019 -G "Visual Studio 16" -DLY_3RDPARTY_PATH=D:\o3de\o3de-packages -DLY_UNITY_BUILD=ON -DLY_PROJECTS=AutomatedTesting

I get the following error message:
CMake Error at CMakeLists.txt:18 (project):
  Generator

    Visual Studio 16 2019

  could not find specified instance of Visual Studio:

    C:/Program Files (x86)/Microsoft Visual Studio/2019/Community

-- Configuring incomplete, errors occurred!

I'm new to using cmake.  Can someone one instruct me how to get it to point to the Professional version of VS2019 instead.

**Comment by RazaelDAbyss** [comment, marked as answer]
I fixed my own problem!
Solution: used Windows Explorer to search contents in the O3DE folder structure for files which contain "Community", specifically looking for CMakeLists.txt and CMakeCache.txt files.  I then edited these files to point to the "Professional" version of Visual Studio 2019.

**Comment by alecazam** [comment]
This is simpler.  You just need to delete CMakeCache.txt.  I hit this as well, but only now have Pro and not Community installed.  I'm not sure which it picks if both are installed.

**Comment by rgba16f** [comment]
In general If you install a new version of visual studio then a full rebuild (delete the build folder) is required as debug file formats and other binary compiled data are not guaranteed to be compatible between different revisions of a compiler.
